src/main/java/org/nrnb/gsoc/enrichment/RequestEngine/HTTPRequestEngine.java

```diff
@@ -148,11 +148,15 @@ public JSONObject makePostRequest(CyNetwork network,String endpoint , Map<String
         httpPost.setHeader("Content-type", "application/json");
         CloseableHttpResponse response = null;
         try {
-            response = httpclient.execute(httpPost);
+            response = ScheduledRequestEngine.sendPostRequestWithTimeLimit(httpclient, httpPost, 20);
```

Comment:
Let's create a constants file and refactor this. For example: `APP_CONSTANTS.TIMEOUT` can be used to store the constant. It makes it easier to maintain the code.

Reply:
Done, but we need to clarify that only global app constants are exclusively kept here. Otherwise storing all constants will create mayhem. Constants required by a single class should be stored in the class itself with `private static final` modifier.